src/suffer.cpp

```diff
@@ -954,65 +1058,41 @@ void suffer::from_sunburn( Character &you )
         affected_part_names.push_back( bp_name );
     }
 
-    std::string all_parts_list = enumerate_as_string( affected_part_names );
+    const std::string all_parts_list = enumerate_as_string( affected_part_names );
 
-    std::string message;
-    if( you.has_trait( trait_ALBINO ) || you.has_effect( effect_datura ) ) {
-        //~ %s is a list of body parts.  The plurality integer is the total
-        //~ number of body parts
-        message = n_gettext( "The sunlight is really irritating your %s.",
-                             "The sunlight is really irritating your %s.",
-                             affected_bodyparts.size() );
-    } else if( you.has_trait( trait_SUNBURN ) ) {
-        //~ %s is a list of body parts.  The plurality integer is the total
-        //~ number of body parts
-        message = n_gettext( "The sunlight burns your %s.",
-                             "The sunlight burns your %s.",
-                             affected_bodyparts.size() );
-    }
-    you.add_msg_if_player( m_bad, message, all_parts_list );
+    // s is a list of body parts.  The plurality integer is the total
+    // number of body parts plus a correction if the eyes are included
+    const int plurality = affected_part_names.size() + contains_eyes;
 
-    // Wake up from skin irritation/burning
-    if( you.has_effect( effect_sleep ) ) {
-        you.wake_up();
-    }
+    auto warn_and_wake_up = [ &you, &all_parts_list, &plurality]
+    ( std::string singular, std::string plural, game_message_type type ) {
+        std::string message = n_gettext( singular.c_str(), plural.c_str(), plurality );
```

Comment:
Sadly this is not how translations work here. You need to add the context to the string literals.

Reply:
What exactly do you mean? As far as I can see it is a normal function taking a singular and a plural string, in addition to an integer describing the amount. The thing with std::string is not that nice and could be replaced with simple char* although that is unrelated (I think)

Reply:
It is a normal function. But it also acts as marker to export the contained string *literals* for translation. You can read about this in the "Developers" section in doc/TRANSLATING.md.

Reply:
Thanks, wasn't aware that a string being a literal made such a difference.